pkg/collect/run.go

```diff
@@ -23,27 +26,62 @@ func Run(c *Collector, runCollector *troubleshootv1beta2.Run) (map[string][]byte
 		return nil, errors.Wrap(err, "failed to create client from config")
 	}
 
-	pod, err := runPod(ctx, client, runCollector, c.Namespace)
-	if err != nil {
-		return nil, errors.Wrap(err, "failed to run pod")
+	// List of Pods that will be created
+	var pods []corev1.Pod
+
+	if runCollector.NodeSelector != nil {
+		// Get all the nodes where a Pod should be running
+		nodes, err := listNodesInSelectors(ctx, client, runCollector.NodeSelector)
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to get the list of nodes matching a nodeSelector")
```

Comment:
```suggestion
			return nil, fmt.Errorf("failed to get the list of nodes matching a nodeSelector, got err: %w", err)
```

You can also use fmt.Error() to generate an error and using the %w flag will wrap the error. I don't know if what you're doing is an anti pattern, probably Simon is better to ask, but figured I'd point it out.

Reply:
tbh, I used what I saw in that code base. I have no clue what is best, or supposed to do. I'm happy to make that change to have a better code quality :smiley:

Reply:
I applied the error handling to use the fmt.Errorf following the reference https://blog.golang.org/go1.13-errors. 
We are currently not handling the type of the Error in the code that receives the error, however it is good to follow the good practices. 

Thank you!